### ISSUE TITLE
def open():

### ISSUE THREAD

**Issue by WK-GiHu** [MTXIII/InvIndex-w-tkinter#4]
@MTXIII 

```
def open():
	top = Toplevel()
	searching = 'Occurences for: ' + Term.get()
	lbl = Label(top, text=searching, width = 60, font = LARGE_FONT).pack()
        listbox = Listbox(top)
        listbox.pack()

        db = bck.Database()
        index = bck.InvertedIndex(db) 
        bck.index_documents(index)

       search_term = Term.get()
       result = bck.index.lookup_query(search_term)    

        for term in result.keys():
            for appearance in result[term]:
                document = db.get(appearance.docId)
                # print(appearance.docId, appearance.frequency)
                # Assuming the Listbox is empty 
                # Maybe requires a NewLine '\n'
                listbox.insert(tk.END, '{}, {}'.format(appearance.docId, appearance.frequency))
```

**Comment by MTXIII** [comment]
oh, so I need to do much more than just call the result from the InvertedIndex.py file.
Updating and testing now.

**Comment by WK-GiHu** [comment]
@MTXIII 

>oh, so I need to do much more than just call the result from the InvertedIndex.py file.

You can strip it down to:
```
def open():
	top = Toplevel()
	searching = 'Occurences for: ' + Term.get()
	lbl = Label(top, text=searching, width = 60, font = LARGE_FONT).pack()
        listbox = Listbox(top)
        listbox.pack()

        for line in bck.index_lookup_query(Term.get())    
                listbox.insert(tk.END, line)
```

This requires a function `index_lookup_query(...` in `bck`, e.g.
```
def index_lookup_query(search_term):
    db = bck.Database()
    index = bck.InvertedIndex(db) 
    bck.index_documents(index)

    result = bck.index.lookup_query(search_term)    
    lines = []
    for key, appearance in result.items():
        lines.append('{}, {}'.format(appearance.docId, appearance.frequency)
    return lines
```

**Comment by MTXIII** [comment]
@WK-GiHu 
I've updated both files as you suggested. 

When I try to run the gui file it fires:
`    listbox = Listbox(top)
                                  ^
IndentationError: unindent does not match any outer indentation level`

from looking at some examples online of tkinter textbox, it appears that the indentation is correct. I've also done a find and replace for tabs, replacing them with spaces just to make sure that's not the issue. 

Trying to find an answer on google / figure it out on my own as this seems pretty simple.. 
but so far no luck.

**Comment by MTXIII** [comment]
Resolved the indentation issue. 
When running the gui file I'm getting:

Exception in Tkinter callback
Traceback (most recent call last):

> Exception in Tkinter callback
> Traceback (most recent call last):
> ...
> ...\gui2.py", line 15, in open
>     for line in bck.index_lookup_query(Term.get()):
> AttributeError: module 'InvertedIndex3' has no attribute 'index_lookup_query'

The `index_lookup_query` function is defined in the file so I'm unsure why the AttributeError is firing. 


Also a separate question, In the InverseIndex3.py file:
I now have duplicate lines of:

>     db = Database()
>     index = InvertedIndex(db) 
>     index_documents(index)
>     result = index.lookup_query(search_term)

(line 97 & line 140) 
Since we've created the index_lookup_query and included those lines inside, can I get rid of them in the main function or are they still needed?


Thank you very much for your help with this @WK-GiHu!
